### ISSUE TITLE
[BUG] Replacing map-styles doesn't change the mapstyle?

### ISSUE THREAD

**Issue by hgnme** [jilleb/mib2-toolbox#150]
I have created and (attempted to have) installed a custom mapstyle for the aus region on my Discover Pro system. After installing and rebooting (power button for 10 seconds), there is no change to the maps that are displayed, either in the dash cluster or the Discover Pro navigation screen.
I have verified the files have actually changed in the MapConfigs folder.

I may be missing a step? Or are tiles being generated and cached? Must I increment the version in version-cfg.txt?

**Describe the bug**
- I have created a customised mapstyle, by editing mapconfig_default_0_DTM and mapconfig_default_kombi_0, etc. for my region (aus)
- I have gzipped the resulting files and copied them back into /mnt/app/navigation/resources/app/vw/aus/MapConfigs/ (using own install.sh, below)
- Upon copying, the filestructure is valid, and the files are confirmed to be actually the new variants
- After a reboot, the Navigation system starts but the map design/style/colouring is unchanged.

**compress.ps1**
```
$files = Get-ChildItem -Path .\ -Filter *.mxf -File -Name
$files | ForEach-Object {
  $gz = '.\' + $_ + '.gz'
  echo $gz
  if (Test-Path $gz) {
    echo 'file exists, deleting'
    Remove-Item $gz
  }
  7z a -tgzip $gz $_
}
```

**install.sh**
```
echo mount app volume as writable.
mount -uw /net/mmx/mnt/app
sleep 0.5

echo copying files from sda0
cp -v -R /fs/sda0/cust_styles/* /mnt/app/navigation/resources/app/vw/aus/MapConfigs/

sleep 0.5
echo remounting app volume as read-only
mount -ur /net/mmx/mnt/app
```

**Expected behavior**
Expect that after replacing the files in MapConfig folder for my region, that the maps change 

**Firmware(please complete the following information):**
 - Brand: VW Golf R 7.5
 - Hardware-type: MIB2 High Discover Pro

**Comment by hgnme** [comment]
I've noticed that the Map version installed on the unit is from 2020 (the latest downloaded from VW Website). However, all data in `/navigation/resources/app/vw/*` is from 2018?

Checking the contexts of the map files in `navigation_styles_VW_AUS.iso` from the 2020 map release shows a revision/version of `10.0.2562`, while the ones in `/mnt/app/navigation/resources/app/vw/aus/` are `10.0.2446`.

I have checked elsewhere on the device for another instance of the navigation files and am unable to find one any, other than the files in the navdb directory (the latest), which are still in their .iso files.

**Comment by vwjap** [comment]
I’m probably 100% wrong on this as I’m only just learning this stuff but isn’t everything in the /mnt directory just The read/write of another place at on time, so when booting up it selects all the stuff needed to read and puts it there, so every time you restart the system it copies the file across from its main directory overwriting your file,

**Comment by olli991** [comment]
The whole function is completly experimental. This is more a theoretical change of files.

But the version could actually be the case. Some time ago I completly replaced everything from audi to my seat folder and it shows up... not good but it was there. My current thinking is that on bootup it checks for the most recent version. If it's in ISO it gathers the files from there. If its is in the folder we extract here so far, it gathers the files from there.

My idea now was to change the file you edited here and increase the version file one digit higher than the one in the ISO. Maybe it then loads from this local storage... not testet in the car so far but you can do so.

**Comment by hgnme** [comment]
I've resolved this. Installing into the directory `/mnt/app/navigation/resources/app/vw/aus/MapConfigs/` is the correct destination (for Australia...).
This issue occurs if a user has updated their map database (navDB) beyond the ones that came stock on the car from factory.

During the map database update, the original files (`/mnt/app/navigation/resources/app/vw/aus/MapConfigs/`) are never actually overwritten. They remain.
The new map files are stored in `/mnt/navdb` (I should stop using mnt paths here...), and remain within their original ISO files.
Each time the nav service starts, it checks all available sources and will load the latest one, regardless of the location. If the latest file is an ISO, it is mounted and the mapStyles data is sourced from it.

Available sources that I know of are:
- `/mnt/app/navigation/resources/app/vw/<region>`
- `/mnt/navdb/mapStyles_<region>`
- `/fs/sda0/`
- `/fs/sdb0/`
The two SD paths will be sourced from if the file `/navigation/LoadStylesFromSDCard` exists.
![image](https://user-images.githubusercontent.com/44087247/109803959-ce5ab100-7c75-11eb-8af0-58934623d4e9.png)

Regardless of the install path, if multiple sources are found (e.g. the updated maps in navDB as well as the default app/vw/region folder), the latest version is used.
The version is specified in `version-cfg.txt`

This issue is resolved by increasing the version in `version-cfg.txt` to later than the version in the updated maps (navDB). This will only load the new styles, the newer maps metadata (streets, poi, etc) is still used.

**Comment by vwjap** [comment]
Ah, gotcha, I thought it was on loadup run a script to place everything we need and put it in the /mnt directory then on shutdown or save rewrite the files listed as changeable in the scripts back tI their original directories

**Comment by hgnme** [comment]
Working well so far. always hated the default styles, so cloning Apple's 👀
![before](https://user-images.githubusercontent.com/44087247/109805557-b6842c80-7c77-11eb-95d8-341f15b88c2f.jpg)
![wip](https://user-images.githubusercontent.com/44087247/109805561-b8e68680-7c77-11eb-9fe3-4609c11d745f.jpg)

Once this is in the active info display, i'll be happy 🙂

**Comment by olli991** [comment]
Good work. So me thinking above was correct. We see how we can automate this for futher releases. Can you maybe attach you current modification. Apple style looks good!

On another quote, what programm is it you are using to make the flow diagram for the functions?

**Comment by olli991** [comment]
As you played with the colorcodes... it is aBGR isn't it?

If you're fine with it we can make your Apple-Style available through the Toolbox? @hgnme

**Comment by vwjap** [comment]
@olli991 have you tried unzipping the zlib file in the startscreens directory?

**Comment by hgnme** [comment]
Yeah it's ABGR but i'm not sure the Alpha is actually used at the moment, I'll try to write up some information on the styles configuration, something for the wiki?

Once it's finished (nightmode,kombi view), i'll stick it in a repo. Happy for it to be included

**Comment by olli991** [comment]
I also tried around and from the existing mapstyles the porsche one looks the best to me. We would like to build a frankenstein skin with the best of all. Including Apple Maps would also be a cool addition to change to.

I also managed to get all this working from SD.. it's quite handy to play around with the skins then.

We are currently planning ISO export scripts and a short insturction on how to simply change the skin to another one... toolbox sd will look like chaos then because mapstyle needs to be in the root so far... let's se what we can do in future releases :-)
